src/query/storage/m3/storage.go

```diff
@@ -85,7 +86,22 @@ func (s *m3storage) Fetch(
 		return nil, err
 	}
 
-	return storage.SeriesIteratorsToFetchResult(raw, s.readWorkerPool, false, s.tagOptions)
+	iters := raw.Iters()
+	accountedIters := make([]encoding.SeriesIterator, len(iters))
+
+	enforcer := options.Enforcer
+	if enforcer == nil {
+		enforcer = cost.NoopEnforcer()
+	}
+	for i, iter := range iters {
+		accountedIters[i] = NewAccountedSeriesIter(iter, enforcer)
+	}
+
+	return storage.SeriesIteratorsToFetchResult(
+		encoding.NewSeriesIterators(accountedIters, raw.Pool()),
```

Comment:
Rather than calling `raw.Pool`, pass in a `*pools.PoolWrapper` into `m3.Storage`, and use `PoolWrapper.IteratorPools().MutableSeriesIterators()`

Reply:
Hmm wouldn't that cause the pools to be different between the two (and thus defeat the point of pooling?). This way seemed more clear to me; since I'm decorating existing iterators, it seemed like it would make sense to use the same pool between the two.

Reply:
Ok tried a different approach here--I'm using the pools that we get back from the namespaces. Lmk if that makes sense!